### PR TITLE
feat: use notation as simp-normal form for BitVec operations

### PR DESCRIPTION
**Pull request by alexkeizer**
Relatively minor change factored out from leanprover-community/mathlib4#5920

### CODE REVIEW

File: Std/Data/BitVec/Lemmas.lean [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import Std.Data.BitVec.Basic
+
+namespace Std.BitVec
+variable {w v : Nat}
+
+/-! We add simp-lemmas that rewrite bitvector operations into the equivalent notation -/
+@[simp] theorem append_eq (x : BitVec w) (y : BitVec v)   : BitVec.append x y = x ++ y        := rfl
+@[simp] theorem shiftLeft_eq (x : BitVec w) (n: Nat)      : BitVec.shiftLeft x n = x <<< n    := rfl
+@[simp] theorem ushiftRight_eq (x : BitVec w) (n: Nat)    : BitVec.ushiftRight x n = x >>> n  := rfl
```
Comment (eric-wieser) [inline Std/Data/BitVec/Lemmas.lean:16]:
```suggestion
@[simp] theorem shiftLeft_eq (x : BitVec w) (n : Nat)     : BitVec.shiftLeft x n = x <<< n    := rfl
@[simp] theorem ushiftRight_eq (x : BitVec w) (n : Nat)   : BitVec.ushiftRight x n = x >>> n  := rfl
```

**Review by eric-wieser — COMMENTED** [review]
Potentially these could even go in `Basic`, since these don't contain any proofs and are really just API. I'll leave that for Joe or Mario to decide.

**Comment by digama0** [comment]
The policy I use is that only equation lemmas and other rfl lemmas go in Basic, and I think this would qualify.

**Comment by eric-wieser** [comment]
@digama0, note you need to click a button for CI to run at all

**Comment by alexkeizer** [comment]
I've just moved the lemmas to the `Basic` file

**Comment by digama0** [comment]
> @digama0, note you need to click a button for CI to run at all

Actually I have to re-click the button after every commit.

**Comment by eric-wieser** [comment]
Once this PR is merged, you won't have to do that for any future PRs from Alex, at least...

**Review by joehendrix — APPROVED** [review]

**Comment by alexkeizer** [comment]
Thanks for the quick review turnaround!

**Pull request merged**